bundle/src/main/java/com/adobe/acs/commons/http/impl/HttpClientFactoryImpl.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2016 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.http.impl;
+
+import com.adobe.acs.commons.http.HttpClientFactory;
+import org.apache.felix.scr.annotations.*;
+import org.apache.http.HttpHost;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.conn.ssl.AllowAllHostnameVerifier;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.osgi.services.HttpClientBuilderFactory;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+
+
+@Component(
+        label = "ACS AEM Commons - Http Components Fluent Executor Factory",
+        description = "ACS AEM Commons - Http Components Fluent Executor Factory",
+        configurationFactory = true, policy = ConfigurationPolicy.REQUIRE)
+@Service
+@Property(label = "Factory Name", description = "Name of this factory", name = "factory.name")
+public class HttpClientFactoryImpl implements HttpClientFactory {
+    public static final boolean DEFAULT_USE_SSL = false;
+
+    public static final boolean DEFAULT_DISABLE_CERT_CHECK = false;
```

Comment:
just a thought, is it better to have final vars in HttpClientFactory? to call HttpClientFactory.DEFAULT_DISABLE_CERT_CHECK rather from implementation HttpClientFactoryImpl.DEFAULT_DISABLE_CERT_CHECK